From 0f0197284574b8447802b95eed0ad6eb816b3810 Mon Sep 17 00:00:00 2001
From: Abishek Ganapathy <quic_abishekg@quicinc.com>
Date: Sat, 17 Feb 2024 00:37:36 +0530
Subject: [PATCH] cfg80211: Add support for STA deletion in 6.6 kernel

Add new case for STA in nl80211_remove_link to delete STA vaps during
deletion.

Change-Id: Ia9f14d0f85c8a2c8b12be38ff5ef6fd804c9961e
Signed-off-by: Abishek Ganapathy <quic_abishekg@quicinc.com>
---
 net/wireless/nl80211.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/net/wireless/nl80211.c b/net/wireless/nl80211.c
index 1cdc3b9f3afd..8ecd617cc75c 100644
--- a/net/wireless/nl80211.c
+++ b/net/wireless/nl80211.c
@@ -16305,6 +16305,7 @@ static int nl80211_remove_link(struct sk_buff *skb, struct genl_info *info)
 
 	switch (wdev->iftype) {
 	case NL80211_IFTYPE_AP:
+	case NL80211_IFTYPE_STATION:
 		break;
 	default:
 		return -EINVAL;
-- 
2.34.1

